File: src/main/java/me/sullivan/glasslang/parser/nodes/StringNode.java
package me.sullivan.glasslang.parser.nodes;

import me.sullivan.glasslang.interpreter.Interpreter;
import me.sullivan.glasslang.interpreter.primitives.Primitive;
import me.sullivan.glasslang.lexer.token.Token;

public class StringNode extends Node {

    public StringNode(Token token)
    {
        super(token);
    }

    @Override
    public String getValue()
    {
        return token.getValue();
    }

    @Override
    public String toString()
    {
        return "String(" + token.getValue() + ")";
    }

    @Override
    public Primitive<?> visitor(Interpreter interpreter)
    {
        return interpreter.visitStringNode(this);
    }
}

File: src/me/sullivan/glasslang/parser/nodes/StringNode.kt
package me.sullivan.glasslang.parser.nodes

import me.sullivan.glasslang.interpreter.Interpreter
import me.sullivan.glasslang.interpreter.primitives.Primitive
import me.sullivan.glasslang.lexer.token.Token

class StringNode(token: Token) : Node(token)
{
    override val value: String
        get() = token.value

    override fun toString(): String
    {
        return "String(" + token.value + ")"
    }

    override fun visitor(interpreter: Interpreter): Primitive<*>
    {
        return interpreter.visitStringNode(this)
    }
}